module/zfs/dmu_objset.c

```diff
@@ -779,18 +791,28 @@ dmu_objset_evict(objset_t *os)
 
 	mutex_enter(&os->os_lock);
 	spa_evicting_os_register(os->os_spa, os);
-	if (list_is_empty(&os->os_dnodes)) {
-		mutex_exit(&os->os_lock);
-		dmu_objset_evict_done(os);
-	} else {
-		mutex_exit(&os->os_lock);
+	mutex_exit(&os->os_lock);
+
+	for (t = 0; t < OS_DNODE_LISTS; t++) {
+		struct dnode_list *dnl = os->os_dnode_list + t;
```

Comment:
[cstyle] The rest of the code bases uses `[t]` rather than `+ t` for this.  So this is preferable for consistency.

```
struct dnode_list *dnl = os->os_dnode_list[t];
```